esphome/core/helpers.cpp

```diff
@@ -551,6 +557,10 @@ void HighFrequencyLoopRequester::stop() {
 bool HighFrequencyLoopRequester::is_high_frequency() { return num_requests > 0; }
 
 void get_mac_address_raw(uint8_t *mac) {  // NOLINT(readability-non-const-parameter)
+#ifdef USE_HOST
+  static const uint8_t esphome_host_mac_addr[6] = {ESPHOME_HOST_MAC_ADDR};
```

Comment:
While I'm here & heckling from the cheap seats... this isn't all that important, but it really irked me to see.

Enclosing `ESPHOME_HOST_MAC_ADDR` in curly brackets like this feels wrong. The usual convention for defined macros is to try have them be something the compiler sees as valid by itself where possible/relevant - that way it's less likely to be misused (when in doubt, don't give yourself a footgun). So I'd expect to see this look like:

```suggestion
  static const uint8_t esphome_host_mac_addr[6] = ESPHOME_HOST_MAC_ADDR;
```

And the curly brackets added when the macro is originally defined (in the call to `add_build_flag` in the Python code)

Reply:
It needs enclosing in braces to make it a valid initialiser. Passing braces via the command line with -D is problematic because of the required escaping and therefore itself prone to misuse.

> The usual convention for defined macros is to try have them be something the compiler sees as valid by itself where possible/relevant

I honestly don't know what that means. A macro is just a textual substitution, its validity is always going to be context dependent.

Would it make you happier if I just left spaces around the braces?

Reply:
I meant in the sense that usually you can put a macro anywhere, without needing to worry about adding anything else to make it a valid token or valid code. Usually. Like you wouldn't define a macro as `hello world"` and expect to need to add a `"` before it every time you use it.

And yea, escaping makes that less nice.... but see my comment about `cg.add_define`.

But as I said, it was just something I saw that irked me (a code smell), not something that's all that important.

Reply:
On further reflection, not only are there implementation difficulties in including the braces in the define, I think that would be wrong. The braces are syntax, not data. Syntax should be fixed and obvious, and not dependent on macro expansion as far as possible.

Reply:
Fair enough